Content.Client/Paint/PaintVisualizerSystem.cs

```diff
@@ -0,0 +1,113 @@
+using System.Linq;
+using Robust.Client.GameObjects;
+using static Robust.Client.GameObjects.SpriteComponent;
+using Content.Shared.Clothing;
+using Content.Shared.Hands;
+using Robust.Client.Graphics;
+using Content.Shared.Paint;
+using Content.Shared.Humanoid;
+using Robust.Shared.Prototypes;
+using System.Reflection.Metadata;
+
+namespace Content.Client.Paint
+{
+    public sealed class PaintedVisualizerSystem : VisualizerSystem<PaintedComponent>
+    {
+        /// <summary>
+        /// Visualizer for Paint which applies a shader and colors the entity.
+        /// </summary>
+
+        [Dependency] private readonly IPrototypeManager _proto = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<PaintedComponent, HeldVisualsUpdatedEvent>(OnHeldVisualsUpdated);
+            SubscribeLocalEvent<PaintedComponent, ComponentShutdown>(OnShutdown);
+            SubscribeLocalEvent<PaintedComponent, EquipmentVisualsUpdatedEvent>(OnEquipmentVisualsUpdated);
+        }
+
+        protected override void OnAppearanceChange(EntityUid uid, PaintedComponent component, ref AppearanceChangeEvent args)
+        {
+            if (args.Sprite == null)
+                return;
+
+            var shader = _proto.Index<ShaderPrototype>("Colored").InstanceUnique();
```

Comment:
Make `"Colored"` a field with `ValidatePrototypeId` attribute

Reply:
The shader now works a little differently (no more post shader). This should essentially be removed.